### <a name="different-exception-handling-for-objectcontextcreatedatabase-and-dbproviderservicescreatedatabase-methods"></a><span data-ttu-id="ef2f7-101">Исключения обрабатываются по-разному для методов ObjectContext.CreateDatabase и DbProviderServices.CreateDatabase</span><span class="sxs-lookup"><span data-stu-id="ef2f7-101">Different exception handling for ObjectContext.CreateDatabase and DbProviderServices.CreateDatabase methods</span></span>

|   |   |
|---|---|
|<span data-ttu-id="ef2f7-102">Подробные сведения</span><span class="sxs-lookup"><span data-stu-id="ef2f7-102">Details</span></span>|<span data-ttu-id="ef2f7-103">Начиная с .NET 4.5, если происходит сбой создания базы данных, методы <code>CreateDatabase</code> будут пытаться удалить пустую базу данных.</span><span class="sxs-lookup"><span data-stu-id="ef2f7-103">Beginning in .NET 4.5, if database creation fails, <code>CreateDatabase</code> methods will attempt to drop the empty database.</span></span> <span data-ttu-id="ef2f7-104">Если эта операция выполняется успешно, будет распространяться исходное исключение <xref:System.Data.SqlClient.SqlException?displayProperty=name> (вместо <xref:System.InvalidOperationException?displayProperty=name>, которое всегда создавалось в .NET 4.0).</span><span class="sxs-lookup"><span data-stu-id="ef2f7-104">If that operation succeeds, the original <xref:System.Data.SqlClient.SqlException?displayProperty=name> will be propagated (instead of the <xref:System.InvalidOperationException?displayProperty=name> that was always thrown in .NET 4.0)</span></span>|
|<span data-ttu-id="ef2f7-105">Предложение</span><span class="sxs-lookup"><span data-stu-id="ef2f7-105">Suggestion</span></span>|<span data-ttu-id="ef2f7-106">При перехвате <xref:System.InvalidOperationException?displayProperty=name> во время выполнения <xref:System.Data.Objects.ObjectContext.CreateDatabase> или <xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)> теперь также должны обрабатываться исключения SQLException.</span><span class="sxs-lookup"><span data-stu-id="ef2f7-106">When catching an <xref:System.InvalidOperationException?displayProperty=name> while executing <xref:System.Data.Objects.ObjectContext.CreateDatabase> or <xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)>, SQLExceptions should now also be caught.</span></span>|
|<span data-ttu-id="ef2f7-107">Область</span><span class="sxs-lookup"><span data-stu-id="ef2f7-107">Scope</span></span>|<span data-ttu-id="ef2f7-108">Дополнительный номер</span><span class="sxs-lookup"><span data-stu-id="ef2f7-108">Minor</span></span>|
|<span data-ttu-id="ef2f7-109">Версия</span><span class="sxs-lookup"><span data-stu-id="ef2f7-109">Version</span></span>|<span data-ttu-id="ef2f7-110">4.5</span><span class="sxs-lookup"><span data-stu-id="ef2f7-110">4.5</span></span>|
|<span data-ttu-id="ef2f7-111">Тип</span><span class="sxs-lookup"><span data-stu-id="ef2f7-111">Type</span></span>|<span data-ttu-id="ef2f7-112">Среда выполнения</span><span class="sxs-lookup"><span data-stu-id="ef2f7-112">Runtime</span></span>|
|<span data-ttu-id="ef2f7-113">Затронутые API</span><span class="sxs-lookup"><span data-stu-id="ef2f7-113">Affected APIs</span></span>|<ul><li><xref:System.Data.Objects.ObjectContext.CreateDatabase?displayProperty=nameWithType></li><li><xref:System.Data.Common.DbProviderServices.CreateDatabase(System.Data.Common.DbConnection,System.Nullable{System.Int32},System.Data.Metadata.Edm.StoreItemCollection)?displayProperty=nameWithType></li></ul>|

